pkg/cli/cli.go

```diff
@@ -207,6 +273,26 @@ func (c cli) validate() error {
 						pluginName, projectVersion, err)
 				}
 			}
+			// Check for duplicate plugin names. Names outside of a version can
```

Comment:
A couple lines above this, we are validating that the returned supported project versions are valid, but that function is being called after `WithPlugins` has already been evaluated. Should we do the check inside `WithPlugins` instead of here? Additionally, as a plugin can be stored in different scaffolding versions, this check may be evaluated multiple times if we leave it here.

Reply:
Either place works for me, however putting as much validation as possible in `cli.validate()` improves readability by semantic encapsulation for a small performance loss from re-evaluating plugin validity.